pkg/securitycontext/util.go

```diff
@@ -232,3 +237,51 @@ func internalSecurityContextFromPodSecurityContext(pod *api.Pod) *api.SecurityCo
 
 	return synthesized
 }
+
+// AddNoNewPrivileges returns if we should add the no_new_privs option. This will return true if:
+// 1) the container is not privileged
+// 2) CAP_SYS_ADMIN is not being added
+// 3) if podSecurityPolicy.DefaultAllowPrivilegeEscalation is:
+//		- nil, then return true if uid != 0
```

Comment:
Yes, if the user isn't set through k8s, the value from the dockerfile is used. I know what you have here is consistent with the proposal, but I'm worried about breaking users with this. It seems like we're already turning it off in most of the cases where it would actually be useful as a security primitive, right? In that case, can we just leave it off in the default case? (i.e. if `pod.Spec..AllowPrivilegeEscalation == nil && psp.DefaultAllowPrivilegEscalation == nil` then the container is run without `no_new_privs`?)